### PR TITLE
Msg workers

### PR DESCRIPTION
**Pull request by dklassen**
Few things going on here. The biggest is spinning up go routines `MessageWorker` to pull from the `robot.ListenChan` and process messages. The naming doesn't make alot of sense anymore but bear with me for the time being.

### CODE REVIEW

**Comment by dklassen** [comment]
cc @bouk

File: slackbot/slack.go [hunk 1]
```diff
@@ -535,25 +531,34 @@ func HerokuPing() {
...
 	}
 
+	robot := NewRobot(origin, apiToken)
 	robot.SlackConnect()
 	robot.DownloadUsersMap()
 	robot.Listen()
 	robot.RegisterCommands(registeredCommands)
 	logrus.Info("Ready and waiting for messages")
 
+	for w := 0; w < 4; w++ {
```
Comment (dklassen) [inline slackbot/slack.go]: make this configurable with a sane default

File: slackbot/slack.go [hunk 2]
```diff
@@ -344,11 +341,14 @@ func (robot *Robot) Listen() {
 			if err != nil {
 				logrus.Error("Error receiving over websocket: ", err.Error())
 			}
+
+			if msg.isMessage() != true {
```
Comment (bouk) [inline slackbot/slack.go]: Don't need !=

Comment (bouk) [reply slackbot/slack.go]: Do ! Instead

File: slackbot/slack.go [hunk 3]
```diff
@@ -535,25 +531,34 @@ func HerokuPing() {
 	}
 }
 
-func (robot Robot) Run() {
+func MessageWorker(robot *Robot) {
+	for {
```
Comment (bouk) [inline slackbot/slack.go]: You should use a range here, so this worker stops when the channel is closed (even if you don't care about that, it's better)

**Comment by bouk** [comment]
You should make the robot methods have a pointer receiver, instead of a value receiver. Right now go needs to copy the object before calling the method

**Comment by dklassen** [comment]
That much I understand and will clean up. I was using the pointer receiver for objects that get mutated. You suggest just using pointer receivers consistently to save the copying?

**Comment by bouk** [comment]
Yes, in general you always use a pointer receiver if the type is a struct

**Pull request merged**